code/modules/antagonists/vampire/vampire_powers/gargantua_powers.dm

```diff
@@ -201,3 +203,88 @@
+
+/obj/effect/proc_holder/spell/vampire/arena/create_new_targeting()
+	var/datum/spell_targeting/click/T = new
+	T.click_radius = 0
+	T.allowed_type = /mob/living/carbon/human
+	return T
+
+/obj/effect/proc_holder/spell/vampire/arena/cast(list/targets, mob/living/user)
+	if(!targets)
+		return
+
+	if(timer) // Recast to dispell the wall and buff early
+		dispell(user)
+		return
+
+	// First we leap towards the enemy target
+	user.add_stun_absorption("gargantua", INFINITY, 2) // We temporarily make the gargantua immune to stuns to stop any matrix fuckery from happening
+	animate(user, 1 SECONDS, pixel_z = 64, flags = ANIMATION_RELATIVE, easing = SINE_EASING|EASE_OUT)
+	addtimer(CALLBACK(user, user.spin(12, 1), 3, 2), 0.3 SECONDS)
+	var/angle = get_angle(user, targets[1]) + 180
+	user.transform = user.transform.Turn(angle)
+	for(var/i in 1 to 10)
+		var/move_dir = get_dir(user, targets[1])
+		user.forceMove(get_step(user, move_dir))
+		if(get_turf(user) == get_turf(targets[1]))
+			user.remove_stun_absorption("gargantua")
+			user.set_body_position(STANDING_UP)
+			user.transform = 0
+			break
+		sleep(1)
+	animate(user, 0.2 SECONDS, pixel_z = -64, flags = ANIMATION_RELATIVE, easing = SINE_EASING|EASE_IN)
+	// They get a cool soundeffect and a visual, as a treat
+
+	playsound(get_turf(user), 'sound/effects/meteorimpact.ogg', 100, TRUE)
+	new /obj/effect/temp_visual/stomp(get_turf(user))
+
+	// Now we build the arena and give the caster the buff
+
+	user.apply_status_effect(STATUS_EFFECT_VAMPIRE_GLADIATOR)
+	spell_active = TRUE
+	timer = addtimer(CALLBACK(src, PROC_REF(dispell), user, TRUE), 30 SECONDS, TIMER_STOPPABLE)
+	RegisterSignal(user, COMSIG_PARENT_QDELETING, PROC_REF(dispell))
+	arena_checks(get_turf(targets[1]), user)
+
+/obj/effect/proc_holder/spell/vampire/arena/proc/arena_checks(turf/target_turf, mob/living/user)
+	if(!spell_active || QDELETED(src))
+		return
+	INVOKE_ASYNC(src, PROC_REF(fighters_check), user)  //Checks to see if our fighters died.
+	INVOKE_ASYNC(src, PROC_REF(arena_trap), target_turf)  //Gets another arena trap queued up for when this one runs out.
+	addtimer(CALLBACK(src, PROC_REF(arena_checks), target_turf, user), 5 SECONDS)
+
+/obj/effect/proc_holder/spell/vampire/arena/proc/arena_trap(turf/target_turf)
+	for(var/tumor_range_turfs in circle_edge_turfs(target_turf, ARENA_SIZE))
+		tumor_range_turfs = new /obj/effect/temp_visual/elite_tumor_wall(tumor_range_turfs, src)
+		all_temp_walls += tumor_range_turfs
+
+/obj/effect/proc_holder/spell/vampire/arena/proc/fighters_check(mob/living/user)
+	if(QDELETED(user) || user.stat == DEAD)
+		dispell(user)
+		return
+
+/obj/effect/proc_holder/spell/vampire/arena/proc/dispell(mob/living/user)
+	spell_active = FALSE
+	if(timer)
+		deltimer(timer)
+		timer = null
+	QDEL_LIST_CONTENTS(all_temp_walls)
+	cooldown_handler.start_recharge()
+	user.remove_status_effect(STATUS_EFFECT_VAMPIRE_GLADIATOR)
+	visible_message("<span class='warning'>The arena begins to dissipate.</span>")
```

Comment:
I don't know if this visible message will work if it's on this abstract proc holder.